variable/string.cpp

```diff
@@ -26,19 +26,24 @@ std::ostream &operator<<(std::ostream &os, const Variable &variable) {
   return os << VariableConstView(variable);
 }
 
+namespace {
 constexpr const char *tab = "  ";
 
-std::string make_dims_labels(const VariableConstView &variable,
-                             const Dimensions &datasetDims) {
+std::string make_dims_labels(
+    const VariableConstView &variable,
+    const std::optional<std::reference_wrapper<const Dimensions>> datasetDims) {
   const auto &dims = variable.dims();
   if (dims.empty())
     return "()";
   std::string diminfo = "(";
   for (const auto dim : dims.labels()) {
     diminfo += to_string(dim);
-    if ((datasetDims.contains(dim) && (datasetDims[dim] + 1 == dims[dim])) ||
-        (datasetDims.empty() && dims[dim] == 2))
-      diminfo += " [bin-edge]";
+    if (datasetDims) {
+      if ((datasetDims->get().contains(dim) &&
+           (datasetDims->get()[dim] + 1 == dims[dim])) ||
+          (datasetDims->get().empty() && dims[dim] == 2))
```

Comment:
What about, e.g., data depending on X, but `dims={y:2}`? Should we get for `!contains(dim)` rather than than `empty()`? Or change the whole thing to

```cpp
if (((datasetDims.contains(dim) ? datasetDims->get()[dim] : 1) + 1 == dims[dim]))
```

Reply:
Good point! Your suggestion seems to work